tests/lib/hardware/hardware_base.py

```diff
@@ -79,6 +80,17 @@ def generate_keys(self):
             "%s%s_key" % (config.CLUSTER_PREFIX, self.hardware_uuid)
         self.pubkey = "%s %s" % (key.get_name(), key.get_base64())
 
+    def ssh_agent(self):
```

Comment:
why is `ssh_agent()` in the base class? it's until now only used in caasp so please move it there